server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -251,12 +261,18 @@ public void parse(
                                 .version(version).versionType(versionType)
                                 .create(true).setPipeline(pipeline).setIfSeqNo(ifSeqNo).setIfPrimaryTerm(ifPrimaryTerm)
                                 .source(sliceTrimmingCarriageReturn(data, from, nextMarker, xContentType), xContentType)
+                                .setMappingHints(mappingHints)
                                 .setRequireAlias(requireAlias), type);
                     } else if ("update".equals(action)) {
                         if (version != Versions.MATCH_ANY || versionType != VersionType.INTERNAL) {
                             throw new IllegalArgumentException("Update requests do not support versioning. " +
                                     "Please use `if_seq_no` and `if_primary_term` instead");
                         }
+                        // TODO: support dynamic mapping hints for update requests
```

Comment:
is the plan to work on this as a follow-up?

Reply:
Yes, I will work on a follow-up for this.